pandas/core/indexes/category.py

```diff
@@ -704,3 +696,18 @@ def _delegate_method(self, name: str, *args, **kwargs):
         if is_scalar(res):
             return res
         return CategoricalIndex(res, name=self.name)
+
+
+def check_indexing_method(self, method):
```

Comment:
why aren't you using pandas.core.missing.clean_fill_method?

Reply:
because thats checking that the method is valid, whereas this is raising whenever the method is non-None.

Reply:
ok seems an odd place to put this inside a sub-class of index, i would locate elsewhere, maybe pandas.core.indexes.api (which btw we should rename to something better)

Reply:
its definitely odd.  we could put it in ExtensionIndex and then have DatetimeLikeIndex override it

Reply:
ok thats a bit better place

Reply:
updated+green

We're about to simplify the tar out of CategoricalIndex indexing